content/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-php-project-for-codespaces.md

```diff
@@ -0,0 +1,209 @@
+
+   ```json
+   // For format details, see https://aka.ms/devcontainer.json. For config options, see the
+   // README at: https://github.com/devcontainers/templates/tree/main/src/php
+   {
+     "name": "PHP",
+     // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+     "image": "mcr.microsoft.com/devcontainers/php:0-8.2",
+
+     // Features to add to the dev container. More info: https://containers.dev/features.
+     // "features": {},
+
+     // Configure tool-specific properties.
+     "customizations": {
+       // Configure properties specific to VS Code.
+       "vscode": {
+         "settings": {},
+         "extensions": [
+           "streetsidesoftware.code-spell-checker",
+           "ikappas.composer"
+         ]
+       }
+     },
+
+     // Use 'forwardPorts' to make a list of ports inside the container available locally.
+     // "forwardPorts": [8000],
+
+     // Use 'portsAttributes' to set default properties for specific forwarded ports. More info: https://code.visualstudio.com/docs/remote/devcontainerjson-reference.
+     "portsAttributes": {
+       "8000": {
+         "label": "Hello Remote World",
+         "onAutoForward": "notify"
+       }
+     },
+
+     // Use 'postCreateCommand' to run commands after the container is created.
+     "postCreateCommand": "composer install",
+
+     // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
+     // "remoteUser": "root"
+   }
+   ```
+
+{% data reusables.codespaces.save-changes %}
+{% data reusables.codespaces.rebuild-command %}
+{% indented_data_reference reusables.codespaces.rebuild-reason %}
+
+   After the dev container is rebuilt, and your codespace becomes available again, the `postCreateCommand` will have been run, installing your Composer dependencies, and the "Composer" extension will be available for use.
+
+## Step 4: Run your application
+
+In the previous section, you used the `postCreateCommand` to install a set of packages via the `composer install` command. With the dependencies now installed, you can run the application.
+
+1. In the Terminal of your codespace, enter `apache2ctl start`.
+
+   <!-- ![Screenshot of running "npm start" in the Terminal. The final output reads "Running on http://0.0.0.0:3000."](/assets/images/help/codespaces/codespaces-npmstart.png) -->
+
+1. When your project starts, you should see a "toast" notification message at the bottom right corner of {% data variables.product.prodname_vscode_shortname %}, telling you that your application is available on a forwarded port. To view the running application, click **Open in Browser**.
+
+   <!-- ![Screenshot of the port forwarding message, reading "Your application running on port 3000 is available." The "Open in Browser" button is also shown.](/assets/images/help/codespaces/codespaces-port3000-toast.png) -->
```

Comment:
When I do this I get: 

```
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.0.1. Set the 'ServerName' directive globally to suppress this message
(13)Permission denied: AH00072: make_sock: could not bind to address [::]:80
(13)Permission denied: AH00072: make_sock: could not bind to address 0.0.0.0:80
no listening sockets available, shutting down
AH00015: Unable to open logs
Action 'start' failed.
The Apache error log may have more information.
```

I tried switching to root and adding "ServerName 127.0.0.1" to `/etc/apache2/apache2.conf` but I get further errors. 

We need to make sure the reader can step through these instructions easily and access a running app in the browser at the end, so this is going to need further work before we can add this article.